### PR TITLE
Math Magicians: refactor with hooks

### PR DESCRIPTION
**Pull request by lu-jim**
Final PR of the project for this block.
## Changes:

- Refactored the Calculator component (components/Calculator.js) from a class-based component to a functional component with hooks.
- Verified the application behaves as expected after the refactor.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,28 +1,27 @@
-import React from 'react';
```
Comment (D4nielJ) [inline src/components/Calculator.js:1]: - [ ] Please do not include `.babelrc`, `eslint.json`, and `.stylelintrc.json` files in `.gitignore`, as these files after they're installed extend the initial config from create-react-app and become important to compile the project.

image

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
-      <>
-        <Screen display={this.state} />
-        <Keys state={this.state} onClick={this.onClick} />
-      </>
-    );
-  }
-}
+  onClick.PropTypes = {
+    value: PropTypes.string.isRequired,
+  };
```
Comment (D4nielJ) [inline src/components/Calculator.js:17]: [OPTIONAL] I think you might have misused PropTypes here. PropTypes are made to indicate react what are the type of the props of a component, but onClick is a function, not a component.

File: src/components/Screen.js [hunk 3]
```diff
@@ -1,25 +1,17 @@
...
+    <div className="screen">
+      {' '}
+      <div>{operation || ''}</div>
+      {' '}
+      {next || total || '0' }
+    </div>
+  );
+};
 
 Screen.propTypes = {
```
Comment (D4nielJ) [inline src/components/Screen.js:16]: You use prop-types correctly here :sweat_smile:

**Review by D4nielJ — APPROVED** [review]
## Hello @lu-jim, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :fire: :rocket: 
- Your calculator works correctly after the refactor! :sparkles: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 2 above)
Comment (lu-jim) [reply src/components/Calculator.js:17]: Thanks! It was a mental typo. Fixed in the next commit.

File: src/components/Calculator.js (hunk 1 above)
Comment (lu-jim) [reply src/components/Calculator.js:1]: Fixed!

**Comment by Davidosky007** [comment]
## Hi @lu-jim 👋, 

## Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 


thumbs-up-gif

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**